test/cheap_ruler.cpp

```diff
@@ -39,6 +39,13 @@ TEST_F(CheapRulerTest, distanceInMiles) {
     assertErr(d / d2, 1.609344, 1e-12);
 }
 
+TEST_F(CheapRulerTest, distanceFiji) {
+    cr::CheapRuler ruler(-16.7);
+    auto const d = ruler.distance({ -179.9854832, -16.7813260 }, { 179.8430685, -16.6728974 });
+
+    assertErr(d, 21.8710, 1.0E-3);
+}
+
```

Comment:
I've added a bunch of tests for all affected cases in JS — would be nice to cover them equivalently here too.